utils/data-migration/tfm/migrate-gef-deals.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * DATA MIGRATION
+ * **************
+ * Purpose of this script is to migrate GEF deals to TFM.
+ * Since GEF deals are not subjected to execution on Workflow.
+ */
+const {
+  getCollection,
+  disconnect,
+} = require('../helpers/database');
+const {
+  getDDMMYYYY,
+  epochInSeconds,
+} = require('../helpers/date');
+const {
+  getDealTfmStage,
+} = require('../helpers/deal');
+const { ObjectId } = require('mongodb');
+const CONSTANTS = require('../constant');
+
+const version = '0.0.1';
+
+// ******************** DEALS *************************
+
+/**
+ * Return all the portal deals, without (default) or with filter specified.
+ * @param {Object} filter Mongo filter
+ * @returns {Object} Collection object
+ */
+const getDeals = (filter = null) => getCollection(CONSTANTS.DATABASE.TABLES.DEAL, filter);
+
+/**
+ * Returns all TFM deals.
+ * @param {Object} filter Mongo filter
+ * @returns {Object} Collection object
+ */
+const getTfmDeals = (filter = null) => getCollection(CONSTANTS.DATABASE.TABLES.TFM_DEAL, filter);
+
+/**
+ * Save deal to `tfm-deals` collection.
+ * @param {Object} deal Deal object
+ * @returns {Boolean} Execution status
+ */
+const setTfmDeal = (deal) => new Promise((response, reject) => {
+  getCollection(CONSTANTS.DATABASE.TABLES.TFM_DEAL, null, true)
+    .then((collection) => collection.insertOne(deal))
+    .then((result) => response(result))
+    .catch((error) => reject(error));
+});
+
+/**
+ * Construct deal's `dealSnapshot` object.
+ * @param {Object} deal Deal object
+ * @returns {Object} dealSnapshot Deal snapshot
+ */
+const buildDealsnapshot = (deal) => (
+  {
+    dealSnapshot: {
+      ...deal,
```

Comment:
you probably have this in your list as well, but I'll leave it here as a reference:
you'll need the `_id` and `dealSnapshot._id` as a MongoDB object. `getCollections` will return it as a string only.
you'll also need inside the `dealSnapshot.facilities` object the entire facility - not just the `id` of it